### PR TITLE
[MAC manager] Improve API consistency and code refactor

### PR DESCRIPTION
**Pull request by kimeunju108**
The MAC address Manager creates and allocate MAC address for each port. Also, it manages MAC address pool and MAC address ranges, which are used when it creates MAC addresses. 
Here is the brief description of MAC manager:

* MAC address APIs: create, get, allocate, update, release

* MAC range APIs: create, get, allocate, update, release

* Schema update: mac_state, mac_range, mac_pool

* Service codes: an inteface MacService and its implementation MacServiceImpl

* Code structure: package path update for common lib, macmanager

* Deployment: Update Application.properties and pom file

### CODE REVIEW

**Comment by xieus** [comment]
We have merge conflicts here.

**Comment by xieus** [comment]
> We have merge conflicts here.

Hi @kimeunju108 , didn't see update since the last check-in. Could you push your latest change here? Thanks.

File: lib/src/main/java/com/futurewei/alcor/common/db/ignite/IgniteConfiguration.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public IgniteClient igniteClientInstance() {
             logger.log(Level.WARNING, "Unexpected failure:" + e.getMessage());
         }
 
-        Assert.notNull(igniteClient, "IgniteClient is null");
+        //Assert.notNull(igniteClient, "IgniteClient is null");
```
[the remark this answers is not in the document]
Comment (kimeunju108) [reply lib/src/main/java/com/futurewei/alcor/common/db/ignite/IgniteConfiguration.java:83]: I removed it. It created assertion error while testing. Now it is fine.

File: lib/src/main/java/com/futurewei/alcor/common/repo/ICacheRepository.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 package com.futurewei.alcor.common.repo;
 
-import com.futurewei.alcor.common.exception.CacheException;
+import com.futurewei.alcor.common.db.CacheException;
```
[the remark this answers is not in the document]
Comment (kimeunju108) [reply lib/src/main/java/com/futurewei/alcor/common/repo/ICacheRepository.java:19]: No, I followed the our teams' common change: some classes wee moved (common.exception ==> common.db)

File: services/mac_manager/src/main/java/com/futurewei/alcor/macmanager/entity/MacState.java [hunk 3]
```diff
@@ -35,23 +35,23 @@
     @JsonProperty("port_id")
     private String portId;
 
-    @JsonProperty("active")
-    private String active;
+    @JsonProperty("state")
+    private String state;
```
[the remark this answers is not in the document]
Comment (kimeunju108) [reply services/mac_manager/src/main/java/com/futurewei/alcor/macmanager/entity/MacState.java:39]: Actually I agree this should be enum type. Already I tried but it caused json error. I will try it again.

File: lib/src/main/java/com/futurewei/alcor/common/repo/ICacheRepository.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (xieus) [reply lib/src/main/java/com/futurewei/alcor/common/repo/ICacheRepository.java:19]: Then it should be fine. Thanks.

**Pull request merged**